chain/network/src/peer_manager/peer_store.rs

```diff
@@ -124,6 +114,24 @@ impl PeerStore {
                 status,
             };
 
+            let is_blacklisted =
+                peer_state.peer_info.addr.as_ref().map_or(false, |addr| blacklist.contains(addr));
+            if is_blacklisted {
+                peers_to_delete.push((peer_id, peer_state));
+            } else {
+                peers_to_keep.push((peer_id, peer_state));
+            }
+        }
+
+        for (_, peer_state) in peers_to_delete.iter() {
+            info!(target: "network", "Removing {:?} because address is blacklisted", peer_state.peer_info);
```

Comment:
Perhaps move this log line when we check `if is_blacklisted`.  This way `peers_to_delete` could just contain `peer_id` and we won’t need to do mapping when calling `delete_from_db` below.